<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsPage</class>
 <widget class="QWidget" name="SettingsPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget#SettingsPage {
    background-color: #404040;
}

QGroupBox {
    background-color: #505050;
    border: 2px solid #666666;
    border-radius: 8px;
    color: #ffffff;
    font-size: 14px;
    font-weight: bold;
    margin-top: 10px;
    padding-top: 5px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 8px 0 8px;
}

QLabel {
    color: #ffffff;
    font-size: 13px;
}

QLineEdit {
    background-color: #555555;
    border: 2px solid #666666;
    border-radius: 6px;
    color: #ffffff;
    padding: 6px 10px;
    font-size: 13px;
}

QLineEdit:focus {
    border: 2px solid #3498db;
    background-color: #606060;
}

QLineEdit:read-only {
    background-color: #454545;
    color: #cccccc;
}

QComboBox {
    background-color: #555555;
    border: 2px solid #666666;
    border-radius: 6px;
    color: #ffffff;
    padding: 6px 10px;
    font-size: 13px;
}

QComboBox:hover {
    border: 2px solid #777777;
    background-color: #606060;
}

QComboBox::drop-down {
    border: none;
    width: 20px;
}

QComboBox::down-arrow {
    image: url(:/icons/arrow_down.png);
    width: 12px;
    height: 12px;
}

QSpinBox {
    background-color: #555555;
    border: 2px solid #666666;
    border-radius: 6px;
    color: #ffffff;
    padding: 6px 10px;
    font-size: 13px;
}

QSpinBox:focus {
    border: 2px solid #3498db;
    background-color: #606060;
}

QCheckBox {
    color: #ffffff;
    font-size: 13px;
    spacing: 8px;
}

QCheckBox::indicator {
    width: 16px;
    height: 16px;
}

QCheckBox::indicator:unchecked {
    background-color: #555555;
    border: 2px solid #666666;
    border-radius: 3px;
}

QCheckBox::indicator:checked {
    background-color: #3498db;
    border: 2px solid #3498db;
    border-radius: 3px;
}

QPushButton {
    background-color: #606060;
    border: 2px solid #777777;
    border-radius: 6px;
    color: #ffffff;
    font-size: 13px;
    padding: 8px 16px;
    min-width: 80px;
}

QPushButton:hover {
    background-color: #707070;
    border: 2px solid #888888;
}

QPushButton:pressed {
    background-color: #505050;
    border: 2px solid #666666;
}

QPushButton#applyButton, QPushButton#okButton {
    background-color: #3498db;
    border: 2px solid #3498db;
    font-weight: bold;
}

QPushButton#applyButton:hover, QPushButton#okButton:hover {
    background-color: #2980b9;
    border: 2px solid #2980b9;
}

QPushButton#applyButton:pressed, QPushButton#okButton:pressed {
    background-color: #21618c;
    border: 2px solid #21618c;
}

QPushButton#resetButton {
    background-color: #e74c3c;
    border: 2px solid #e74c3c;
}

QPushButton#resetButton:hover {
    background-color: #c0392b;
    border: 2px solid #c0392b;
}

QPushButton#resetButton:pressed {
    background-color: #a93226;
    border: 2px solid #a93226;
}

QScrollArea {
    background-color: transparent;
    border: none;
}

QScrollArea > QWidget > QWidget {
    background-color: transparent;
}

QWidget#scrollAreaWidget {
    background-color: transparent;
}

QScrollBar:vertical {
    background-color: #505050;
    width: 12px;
    border-radius: 6px;
}

QScrollBar::handle:vertical {
    background-color: #777777;
    border-radius: 6px;
    min-height: 20px;
}

QScrollBar::handle:vertical:hover {
    background-color: #888888;
}

QScrollBar::add-line:vertical, QScrollBar::sub-line:vertical {
    height: 0px;
}</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <property name="spacing">
    <number>15</number>
   </property>
   <property name="leftMargin">
    <number>20</number>
   </property>
   <property name="topMargin">
    <number>20</number>
   </property>
   <property name="rightMargin">
    <number>20</number>
   </property>
   <property name="bottomMargin">
    <number>20</number>
   </property>
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <widget class="QWidget" name="scrollAreaWidget">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>580</width>
        <height>650</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="contentLayout">
       <property name="spacing">
        <number>15</number>
       </property>
       <item>
        <widget class="QGroupBox" name="generalGroup">
         <property name="title">
          <string>General</string>
         </property>
         <layout class="QGridLayout" name="generalLayout">
          <property name="spacing">
           <number>10</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="themeLabel">
            <property name="text">
             <string>Theme:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="themeCombo">
            <item>
             <property name="text">
              <string>Light</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Dark</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>System</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="languageLabel">
            <property name="text">
             <string>Language:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="languageCombo">
            <item>
             <property name="text">
              <string>English</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Chinese</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Japanese</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="accentColorLabel">
            <property name="text">
             <string>Accent Color:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QPushButton" name="accentColorButton">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>Select Color</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="networkGroup">
         <property name="title">
          <string>Network</string>
         </property>
         <layout class="QGridLayout" name="networkLayout">
          <property name="spacing">
           <number>10</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="apiUrlLabel">
            <property name="text">
             <string>API Base URL:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="apiUrlEdit">
            <property name="placeholderText">
             <string>https://api.bilibili.com</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="timeoutLabel">
            <property name="text">
             <string>Timeout (ms):</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="timeoutSpinBox">
            <property name="suffix">
             <string> ms</string>
            </property>
            <property name="minimum">
             <number>1000</number>
            </property>
            <property name="maximum">
             <number>30000</number>
            </property>
            <property name="singleStep">
             <number>1000</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="proxyUrlLabel">
            <property name="text">
             <string>Proxy URL:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="proxyUrlEdit">
            <property name="placeholderText">
             <string>http://proxy.example.com:8080</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="pathsGroup">
         <property name="title">
          <string>Directories</string>
         </property>
         <layout class="QGridLayout" name="pathsLayout">
          <property name="spacing">
           <number>10</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="playlistDirLabel">
            <property name="text">
             <string>Playlists:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="playlistDirEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="playlistDirButton">
            <property name="text">
             <string>Browse...</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="themeDirLabel">
            <property name="text">
             <string>Themes:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="themeDirEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="themeDirButton">
            <property name="text">
             <string>Browse...</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="audioCacheDirLabel">
            <property name="text">
             <string>Audio Cache:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="audioCacheDirEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QPushButton" name="audioCacheDirButton">
            <property name="text">
             <string>Browse...</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="coverCacheDirLabel">
            <property name="text">
             <string>Cover Cache:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="coverCacheDirEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QPushButton" name="coverCacheDirButton">
            <property name="text">
             <string>Browse...</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="playlistGroup">
         <property name="title">
          <string>Playlists</string>
         </property>
         <layout class="QGridLayout" name="playlistLayout">
          <property name="spacing">
           <number>10</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="autoSaveIntervalLabel">
            <property name="text">
             <string>Auto-save Interval:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="autoSaveIntervalSpinBox">
            <property name="suffix">
             <string> minutes</string>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>60</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
            <property name="value">
             <number>5</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QCheckBox" name="autoSaveCheckBox">
            <property name="text">
             <string>Enable auto-save</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="advancedGroup">
         <property name="title">
          <string>Advanced</string>
         </property>
         <layout class="QHBoxLayout" name="advancedLayout">
          <property name="spacing">
           <number>10</number>
          </property>
          <item>
           <widget class="QPushButton" name="exportButton">
            <property name="text">
             <string>Export Settings...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="importButton">
            <property name="text">
             <string>Import Settings...</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="advancedSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="buttonLayout">
     <property name="spacing">
      <number>10</number>
     </property>
     <item>
      <spacer name="buttonSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="resetButton">
       <property name="text">
        <string>Reset to Defaults</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="applyButton">
       <property name="text">
        <string>Apply</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="okButton">
       <property name="text">
        <string>OK</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>